pof/src/parse.rs

```diff
@@ -214,33 +256,24 @@ impl<R: Read + Seek> Parser<R> {
                     })?);
                     //println!("{:#?}", eye_points);
                 }
-                b"GPNT" => {
-                    primary_weps = Some(self.read_list(|this| {
-                        this.read_list(|this| {
-                            Ok(WeaponHardpoint {
-                                position: this.read_vec3d()?,
-                                normal: this.read_vec3d()?,
-                                offset: (this.version >= Version::V22_01 || this.version == Version::V21_18)
-                                    .then(|| this.read_f32().unwrap())
-                                    .unwrap_or(0.0),
-                            })
-                        })
-                    })?);
-                    //println!("{:#?}", primary_weps);
-                }
-                b"MPNT" => {
-                    secondary_weps = Some(self.read_list(|this| {
+                b"GPNT" | b"MPNT" => {
+                    let target = if id == b"GPNT" { &mut primary_weps } else { &mut secondary_weps };
+                    assert!(target.is_none());
+                    *target = Some(self.read_list(|this| {
                         this.read_list(|this| {
                             Ok(WeaponHardpoint {
                                 position: this.read_vec3d()?,
                                 normal: this.read_vec3d()?,
-                                offset: (this.version >= Version::V22_01 || this.version == Version::V21_18)
-                                    .then(|| this.read_f32().unwrap())
-                                    .unwrap_or(0.0),
+                                // TODO: document this at https://wiki.hard-light.net/index.php/POF_data_structure
```

Comment:
```
 #ifdef version2201orhigher
   float offset[g,s]
 #endif```
It is documented, unless you mean a more specific versioning for it.

Reply:
I mean that 21.18 is not documented there.